LLT_ALIGN makes Microsoft VC++ unhappy

this macro

```
#define LLT_ALIGN(x, sz) (((x) + (sz-1)) & (-sz))
```

in [support/dtypes.h](https://github.com/JuliaLang/julia/blob/79b241872d55d556189ce2fb4e91157164b2b1a6/src/support/dtypes.h#L160) causes Visual C++ to complain with [the following warning](https://msdn.microsoft.com/en-us/library/4kh09110.aspx)

```
julia.h(750) warning C4146: unary minus operator applied to unsigned type, result still unsigned
```

The macro may likely be right but it unnerves the compiler and because it's in julia.h, it's all over the place littering the output with this warning -- which can be suppressed.  It would be nice to have it not do that by default is there's another way to construct the effect.

[pao: insert pinned SHA link to `dtypes.h`]

Oh right, I don't have a copy of that version of visual studio installed locally. Here it is running on AppVeyor though, not seeing this warning https://ci.appveyor.com/project/tkelman/julia/build/1.0.687

I get one in `c:/projects/julia/src/gc.c(581): warning C4312: 'type cast': conversion from 'unsigned int' to 'void *' of greater size` and a bunch about unrecognized flags. How are you building this? You might be missing some of the `-D` flags that the makefiles are using.

Then it fails to link because I don't have a built copy of LLVM using the right version of visual studio.

--- a/src/support/dtypes.h
+++ b/src/support/dtypes.h
@@ -227,7 +227,7 @@ STATIC_INLINE unsigned int next_power_of_two(unsigned int val) JL_NOTSAFEPOINT
     return val;
 }
 
-#define LLT_ALIGN(x, sz) (((x) + (sz)-1) & ~(sz-1))
+#define LLT_ALIGN(x, sz) (((x) + (sz)-1) & ~((sz)-1))
 
 // branch prediction annotations
 #ifdef __GNUC__

--- a/src/support/dtypes.h
+++ b/src/support/dtypes.h
@@ -227,7 +227,7 @@ STATIC_INLINE unsigned int next_power_of_two(unsigned int val) JL_NOTSAFEPOINT
     return val;
 }
 
-#define LLT_ALIGN(x, sz) (((x) + (sz)-1) & -(sz))
+#define LLT_ALIGN(x, sz) (((x) + (sz)-1) & ~(sz-1))
 
 // branch prediction annotations
 #ifdef __GNUC__

